### PR TITLE
Add tan, arcsin, arccos, arctan to ChainerX

### PR DESCRIPTION
**Pull request by IvanYashchuk**
With reference to #6423 

Changelog:
- Added `tan`, `arcsin`, `arccos`, `arctan` to ChainerX routines.
- Added Python tests.
- Updated docs

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -1649,6 +1649,102 @@ def func(self, xp, a):
...
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'in_dtypes,out_dtype': _in_out_dtypes_math_functions,
+        'input': [-0.75, 0, 0.75],
+        'contiguous': [None, 'C'],
+    })
+    # Special values
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'in_dtypes,out_dtype': _in_out_float_dtypes_math_functions,
+        'input': [1.57, 2, 3.14, float('inf'), -float('inf'), float('nan')],
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Could you make these values more appropriate? (For example, `[1, -1, float('inf'), -float('inf')]` seems better.)

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 2]
```diff
@@ -1649,6 +1649,102 @@ def func(self, xp, a):
...
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: How about `[1, -1, float('inf'), -float('inf')]`?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (IvanYashchuk) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: I agree it is better. I'll make the changes.

Comment (asi1024) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Sorry, I mistaked. `[1, -1, float('inf'), -float('inf'), float('nan')]` is better.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
The test failures seem unrelated to this PR. LGTM! Thank you very much!

**Pull request merged**